Sources/BoundariesTestSupport/Support.swift

```diff
@@ -2,31 +2,57 @@ import Boundaries
 import NonEmpty
 import SnapshotTesting
 
-public final class TestStore<S, A> {
-  public let reducer: Reducer<S, A>
-  public private(set) var history: NonEmptyArray<(message: String, action: A?, state: S, effect: Effect<A>)>
+public final class TestStore<S, E: EffectProtocol> {
```

Comment:
hate to say this, but maybe `TestStore` _inherits_ from `Store` :( everything is the same except for the interpreter...

Reply:
Don't think we need inheritance or even a protocol here. Since `Store` gets configured with an interpreter now, I don't even think we need `TestStore`. We could compose `TestStore` as a wrapper type that holds onto a `Store` and a `history`, or we could have a test store subscriber that manages the history and snapshot assertions against it.

Reply:
down for any of those things. the main things we have to address with the test store is 1) capturing history and 2) interpreting effects synchronously (basically turning `.parallel` into just a `.forEach`.

everything else is the same I believe.